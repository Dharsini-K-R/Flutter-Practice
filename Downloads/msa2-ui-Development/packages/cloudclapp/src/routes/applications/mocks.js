export const dockerImage = [
  {
    id: "",
    name: "node",
    slug: "node",
    type: "image",
    publisher: {
      id: "docker",
      name: "Docker",
    },
    created_at: "2016-06-17T17:45:46.898676Z",
    updated_at: "2022-02-04T06:12:22.541527Z",
    short_description:
      "Node.js is a JavaScript-based platform for server-side and networking applications.",
    source: "store",
    popularity: 0,
    categories: [
      {
        name: "application_infrastructure",
        label: "Application Infrastructure",
      },
    ],
    operating_systems: [
      {
        name: "linux",
        label: "Linux",
      },
    ],
    architectures: [
      {
        name: "arm64",
        label: "ARM 64",
      },
      {
        name: "s390x",
        label: "IBM Z",
      },
      {
        name: "ppc64le",
        label: "PowerPC 64 LE",
      },
      {
        name: "386",
        label: "386",
      },
      {
        name: "arm",
        label: "ARM",
      },
      {
        name: "amd64",
        label: "x86-64",
      },
    ],
    logo_url: {
      large:
        "https://d1q6f0aelx0por.cloudfront.net/product-logos/library-node-logo.png",
      small:
        "https://d1q6f0aelx0por.cloudfront.net/product-logos/library-node-logo.png",
    },
    certification_status: "not_certified",
    star_count: 11111,
    pull_count: "1B+",
    filter_type: "official",
  },
];

export const VMImage = [
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
];
