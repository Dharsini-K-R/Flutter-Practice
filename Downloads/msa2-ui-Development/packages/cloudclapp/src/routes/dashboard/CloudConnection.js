import React, { useState } from "react";
import { useTranslation } from "react-i18next";

import useApi from "cloudclapp/src/hooks/useApi";
import usePostApi from "cloudclapp/src/hooks/usePostApi";
import {
  createCloudConnection,
  updateCloudConnection,
  deleteCloudConnection,
  getCloudConnection,
} from "cloudclapp/src/api/cloud";

import {
  makeStyles,
  Grid,
  Box,
  TextField,
  Button,
  Typography,
} from "@material-ui/core";

import Dialog from "cloudclapp/src/components/dialog/Dialog";
import { useSelector } from "react-redux";
import {
  getCloudVendors,
  getOrganisation,
} from "cloudclapp/src/store/designations";

import { startCase } from "lodash";

import CloudVendorIcon from "cloudclapp/src/components/cloud-vendor-icon";

const useStyles = makeStyles(({ palette, spacing }) => {
  return {
    loadingWrapper: {
      position: "absolute",
      top: 0,
      left: 0,
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      height: "100%",
      width: "100%",
      backgroundColor: palette.background.subTextGrey,
    },
    cloudIcon: {
      width: "52px",
      height: "52px",
      borderRadius: "4px",
      boxShadow: `0px 4px 16px ${palette.background.boxShadow}`,
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
    },
    flexColumn: {
      paddingLeft: "16px",
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      alignItems: "flex-start",
    },
    cloudVendor: {
      fontWeight: "bold",
    },
    fontConnected: {
      color: palette.background.checkGreen,
      fontSize: "12px",
      fontWeight: "600",
    },
    fontSubtext: {
      color: palette.background.subTextGrey,
      fontSize: "12px",
      fontWeight: "600",
    },
    flexSpaceBetween: {
      display: "flex",
      justifyContent: "space-between",
      alignItems: "center",
      paddingTop: "20px",
    },
    formField: {
      margin: `${spacing(1)}px 0`,
      textAlign: "left",
      weight: "100%",
    },
    cloudDesc: {
      color: palette.text.support,
      fontSize: "14px",
      paddingTop: "16px",
      textAlign: "left",
    },
    helperText: {
      color: palette.primary.main,
      cursor: "pointer",
    },
    flexSpaceBtw: {
      display: "flex",
      justifyContent: "space-between",
      alignItems: "center",
      width: "100%",
    },
  };
});

const CloudConnection = ({ onClose, cloudContData }) => {
  const classes = useStyles();
  const { t } = useTranslation();

  const connection = cloudContData.connections[0];
  const hasConnection = Boolean(connection?.connectionName);

  const { id: orgId, prefix } = useSelector(getOrganisation);

  const [credentials, setCredentials] = useState({});

  const cloudVendors = useSelector(getCloudVendors);

  const [isLoading] = useApi(
    getCloudConnection,
    {
      operatorPrefix: prefix,
      vendor: cloudContData.cloudVendor,
      connectionName: connection?.connectionName,
      transforms: [
        (response) => {
          setCredentials(response);
          return response;
        },
      ],
    },
    !hasConnection,
  );

  const [onClickConnect, isConnecting] = usePostApi(
    createCloudConnection,
    {
      orgId,
      vendor: cloudContData.cloudVendor,
      connectionName: `${cloudContData.cloudVendor}-connection-1`,
      credentials,
    },
    onClose,
  );
  const [onClickUpdate, isUpdating] = usePostApi(
    updateCloudConnection,
    {
      orgId,
      vendor: cloudContData.cloudVendor,
      connectionName: connection?.connectionName,
      credentials,
    },
    onClose,
  );

  const [onClickDelete, isDeleting] = usePostApi(
    deleteCloudConnection,
    {
      orgId,
      vendor: cloudContData.cloudVendor,
      connectionName: connection?.connectionName,
    },
    onClose,
  );

  return (
    <Dialog
      maxWidth="md"
      onClose={onClose}
      title={`${cloudContData.vendorDisplayName} ${t("Connections")}`}
      data-testid="manage-connections-dialog"
      disabled={isLoading || isConnecting || isUpdating || isDeleting}
      onExec={hasConnection ? onClickUpdate : onClickConnect}
      execLabel={hasConnection ? t("Update") : t("Connect")}
      extraAction={hasConnection ? onClickDelete : undefined}
      extraLabel={hasConnection ? t("Delete") : undefined}
    >
      <Grid>
        <Box
          sx={{
            display: "flex",
            flexDirection: "row",
            justifyContent: "space-between",
          }}
        >
          <Box
            sx={{
              justifyContent: "flex-start",
              display: "flex",
              flexDirection: "row",
            }}
          >
            <div className={classes.cloudIcon}>
              <CloudVendorIcon vendor={cloudContData.cloudVendor} />
            </div>
            <div className={classes.flexColumn}>
              <div>
                <span
                  className={classes.cloudVendor}
                  id="MANAGE_CLOUD_VENDOR_NAME"
                >
                  {cloudContData.vendorDisplayName}
                </span>
                {hasConnection && " - "}
                <span id="MANAGE_CLOUD_VENDOR_CONNECTION_NAME">
                  {connection?.connectionName}
                </span>
              </div>
              {connection?.connectionStatus === "Connected" ? (
                <div
                  className={classes.fontConnected}
                  id="MANAGE_CLOUD_CONNECTED"
                >
                  {connection?.connectionStatus}
                </div>
              ) : (
                <div
                  className={classes.fontSubtext}
                  id="MANAGE_CLOUD_NOTCONNECTED"
                >
                  {connection?.connectionStatus}
                </div>
              )}
            </div>
          </Box>
          {cloudVendors[cloudContData.cloudVendor]?.helper !== null && (
            <Box
              sx={{
                justifyContent: "flex-start",
                display: "flex",
                flexDirection: "row",
                alignItems: "center",
              }}
            >
              <Button
                variant="outlined"
                color="primary"
                size="small"
                target="_blank"
                id={`MANAGE_CONNECTION_${cloudContData.cloudVendor}_HELPER`}
                href={cloudVendors[cloudContData.cloudVendor]?.helper}
              >
                {t("HELPER")}
              </Button>
            </Box>
          )}
        </Box>
        <div className={classes.cloudDesc}>
          <span>
            {t(
              "Service provider will provide you with a KEY when you create AWS account.",
              { value: cloudContData.vendorDisplayName },
            )}
          </span>{" "}
          {/* <span className={classes.helperText}>
            If you don't have an account, click here to create an account
            {t("click here to create an account")}
          </span> */}
        </div>

        <Box sx={{ display: "flex", flexDirection: "column", pt: 2 }}>
          {cloudVendors[cloudContData.cloudVendor].credentials_required.map(
            (lable, index) => {
              return (
                <Grid className={classes.flexSpaceBtw} key={index}>
                  {" "}
                  <Grid item>
                    <Typography variant="body1" id="MANAGE_CLOUD_LABEL">
                      {startCase(lable)}
                    </Typography>
                  </Grid>
                  <Grid item xs={8}>
                    <TextField
                      value={credentials[lable]}
                      onChange={({ target: { value } }) => {
                        setCredentials({ ...credentials, [lable]: value });
                      }}
                      variant="outlined"
                      id={`MANAGE_CONNECTION_KEY_${index}`}
                      label={null}
                      required
                      className={classes.formField}
                      disabled={
                        isLoading || isConnecting || isUpdating || isDeleting
                      }
                      fullWidth={true}
                      multiline
                      maxRows={3}
                    />
                  </Grid>
                </Grid>
              );
            },
          )}
        </Box>
      </Grid>
    </Dialog>
  );
};

export default CloudConnection;
