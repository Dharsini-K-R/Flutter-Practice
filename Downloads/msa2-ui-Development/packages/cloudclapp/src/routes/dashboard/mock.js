export const organisation = [
  { environmentCount: 20, usersCount: 1, prefix: "00p" },
];

export const environments = [
  {
    status: "Critical",
    version: "V1.1.1",
    user: "Bib Dev",
    env: "Development",
    envData:
      "Development Environment for all new projects and major updates. Nginx and PHP added to the Environment",
    deplymentCount: "2",
    cost: "58.2",
  },
];

export const cloudVendors = {
  aws: {
    displayName: "AWS",
    category: "Public Cloud",
    logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/aws.logo",
    credentials_required: ["aws_access_key", "aws_secret_key"],
    services: {
      eks: {
        displayName: "Elastic Container Service for Kubernetes",
        logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/aws_eks.logo",
        workflow: {
          env:
            "Process/cloudclapp-wf/Provision_an_EKS_Cluster__AWS_/Provision_an_EKS_Cluster__AWS_",
          appsDeployment:
            "Process/cloudclapp-wf/Provision_Apps_EKS_Cluster/Provision_Apps_EKS_Cluster",
          scanApp:
            "Process/cloudclapp-wf/Scan_a_App_Container_/Scan_a_App_Container_",
        },
      },
    },
  },
  azure: {
    displayName: "AZURE",
    category: "Public Cloud",
    logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/azure.logo",
    credentials_required: ["azure_client_id", "azure_client_secret"],
    services: {
      aks: {
        displayName: "Azure Kubernetes Service",
        logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/azure_aks.logo",
        workflow: {
          env:
            "Process/cloudclapp-wf/Provision_an_AKS_Cluster__AZURE_/Provision_an_AKS_Cluster__AZURE_",
          appsDeployment:
            "Process/cloudclapp-wf/Provision_an_AKS_Apps__AZURE_/Provision_an_AKS_Apps__AZURE_",
          scanApp:
            "Process/cloudclapp-wf/Scan_a_App_Container_/Scan_a_App_Container_",
        },
      },
    },
  },
  gcp: {
    displayName: "Google Cloud Platform",
    category: "Public Cloud",
    logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/gcp.logo",
    credentials_required: [
      "gcp_service_account_email",
      "gcp_private_key_id",
      "gcp_private_key",
      "gcp_client_id",
    ],
    services: {
      gke: {
        displayName: "Google Kubernetes Engine",
        logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/gcp_gke.logo",
        workflow: {
          env:
            "Process/cloudclapp-wf/Provision_an_GKE_Cluster__GCP_/Provision_an_GKE_Cluster__GCP_",
          appsDeployment:
            "Process/cloudclapp-wf/Provision_Apps_GKE_Cluster/Provision_Apps_GKE_Cluster",
          scanApp:
            "Process/cloudclapp-wf/Scan_a_App_Container_/Scan_a_App_Container_",
        },
      },
    },
  },
};
