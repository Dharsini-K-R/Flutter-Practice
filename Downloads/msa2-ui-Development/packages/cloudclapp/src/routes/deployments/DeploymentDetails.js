import React from "react";
import PropTypes from "prop-types";
import { useSelector } from "react-redux";
import { Grid, Typography } from "@material-ui/core";
import { StreamIcon, AngleUpIcon, AngleDownIcon } from "react-line-awesome";
import { makeStyles, Tooltip } from "@material-ui/core";
import UserButton from "cloudclapp/src/components/user-button/UserButton";
import { useTranslation } from "react-i18next";
import StatusBadgeIcon from "cloudclapp/src/components/status-badge-icon/StatusBadgeIcon";
import useWorkflowDialog from "cloudclapp/src/hooks/useWorkflowDialog";
import DeploymentsExpandedView from "./DeploymentsExpandedView";
import useToggle from "react-use/lib/useToggle";
import { WORKFLOW_STATUS } from "cloudclapp/src/Constants";
import Process from "msa2-ui/src/services/Process";
import CloudVendor from "cloudclapp/src/services/CloudVendor";
import {
  getCloudVendors,
  getWorkflowByUri,
} from "cloudclapp/src/store/designations";

const useStyles = makeStyles(({ breakpoints, palette }) => {
  return {
    boxWrapper: {
      padding: "10px",
      width: "100%",
      borderBottom: "1px solid #e0e2e7",
    },
    deploymentWrapper: {
      padding: "6px 8px",
      alignItems: "center",
    },
    icon: {
      fontSize: "21px",
    },
    iconText: {
      fontSize: "15px",
      marginLeft: "6px",
    },
    expandCollapseIcon: {
      fontSize: 19,
      verticalAlign: "middle",
      color: palette.common.black,
      cursor: "pointer",
    },
    statusRunning: (colorProps) => ({
      color: colorProps.wfColor,
      marginLeft: "6px",
    }),
    statusReady: {
      color: palette.primary.main,
    },
    workflowIcon: {
      padding: "5px",
      borderRadius: "4px",
      border: "1px solid",
      fontSize: "20px",
      color: palette.primary.main,
      cursor: "pointer",
    },
    userContainer: {
      justifyContent: "flex-start",
      [breakpoints.down("sm")]: {
        justifyContent: "flex-end",
      },
    },
    containerPlacement: {
      justifyContent: "flex-start",
      [breakpoints.up("md")]: {
        justifyContent: "flex-end",
      },
    },
    titleContainer: {
      display: "flex",
      padding: "6px 8px",
      cursor: "pointer",
      "&:hover": {
        backgroundColor: "rgba(56, 64, 82, 0.04)",
      },
    },
  };
});

const DeploymentDetails = ({
  deploymentDetails,
  environment,
  isExpandable,
  onClickCallBack,
}) => {
  const { envUbiqubeId: ubiqubeId } = environment;
  const { t } = useTranslation();
  const cloudVendors = useSelector(getCloudVendors);
  const { appWFUri } = CloudVendor.getWFUrisFromEnvironment(
    environment,
    cloudVendors,
  );

  const [drawerOpen, toggleDrawer] = useToggle(false);

  const deployment = {};

  if (deploymentDetails.variables) {
    deployment.name = deploymentDetails.variables.deployment_name;
    deployment.id = deploymentDetails.variables.SERVICEINSTANCEID;
    deployment.user = deploymentDetails.taskRunner;
    deployment.status = deploymentDetails.status.status;
  } else {
    deployment.name = deploymentDetails.deploymentName;
    deployment.id = deploymentDetails.deploymentId;
    deployment.user = deploymentDetails.deploymentUser;
    deployment.status = deploymentDetails.status;
  }

  const statusObject = WORKFLOW_STATUS[deployment.status];

  const workflow = useSelector(getWorkflowByUri(appWFUri));

  const {
    showWorkflowDialog,
    WorkflowDialog,
    ...workflowDialogProps
  } = useWorkflowDialog({
    workflow,
    ubiqubeId,
    instanceId: deployment.id,
  });

  deployment.displayStatus = statusObject.name;
  const colorProps = { wfColor: statusObject.color };
  const classes = useStyles(colorProps);

  return (
    <Grid container className={classes.boxWrapper}>
      <WorkflowDialog {...workflowDialogProps} />
      <Grid
        container
        item
        lg={3}
        md={4}
        sm={6}
        xs={6}
        className={classes.deploymentWrapper}
        onClick={() => onClickCallBack(deployment.id)}
      >
        <div className={classes.titleContainer}>
          <StatusBadgeIcon
            icon={StreamIcon}
            size="small"
            type="deployment"
            status={deployment.status}
          />
          <Typography
            id={`DEPLOYMENTS_DEPLOYMENT_NAME_${deployment.id}`}
            className={classes.iconText}
          >
            {deployment.name}
          </Typography>
        </div>
      </Grid>
      <Grid
        container
        item
        lg={3}
        md={2}
        sm={6}
        xs={6}
        className={`${classes.userContainer} ${classes.deploymentWrapper}`}
      >
        <UserButton
          id={`DEPLOYMENT_DETAILS_OWNER_BUTTON_${deployment.id}`}
          className={classes.icon}
          username={deployment.user}
          color="dark"
        />
      </Grid>
      <Grid container item lg={6} md={6} sm={12} xs={12} spacing-lg={2}>
        <Grid
          item
          container
          xs
          className={`${classes.deploymentWrapper} ${classes.containerPlacement}`}
        >
          <Typography
            id={`DEPLOYMENTS_DEPLOYMENT_STATUS_${deployment.id}`}
            className={classes.iconText}
          >
            {t("Status")}:
            <span className={classes.statusRunning}>
              {deployment.displayStatus}
            </span>
          </Typography>
          <Grid
            item
            container
            lg={1}
            md={2}
            sm={2}
            xs={1}
            justifyContent="flex-end"
          >
            {workflow &&
              statusObject?.deployActions?.map(
                ({ name, processName, icon: StatusIcon }, index) => {
                  const process = Process.getProcessByLabel(
                    processName,
                    workflow.process,
                  );
                  if (!process) return null;
                  return (
                    <Tooltip title={processName}>
                      <StatusIcon
                        key={name}
                        id={`DEPLOYMENT_DETAILS_STATUS_ICON_${index}`}
                        className={classes.workflowIcon}
                        onClick={() => {
                          showWorkflowDialog(process);
                        }}
                      />
                    </Tooltip>
                  );
                },
              )}
          </Grid>
        </Grid>

        {isExpandable && (
          <Grid
            item
            container
            xs={2}
            sm={1}
            lg={2}
            md={2}
            className={`${classes.deploymentWrapper} ${classes.containerPlacement}`}
          >
            {drawerOpen && (
              <AngleUpIcon
                id={`DEPLOYMENTS_DEPLOYMENT_COLLAPSE_${deployment.id}`}
                className={classes.expandCollapseIcon}
                onClick={toggleDrawer}
              />
            )}
            {!drawerOpen && (
              <AngleDownIcon
                id={`DEPLOYMENTS_DEPLOYMENT_EXPAND_${deployment.id}`}
                className={classes.expandCollapseIcon}
                onClick={toggleDrawer}
              />
            )}
          </Grid>
        )}
      </Grid>
      {drawerOpen && (
        <Grid container>
          <DeploymentsExpandedView
            instanceId={deployment.id}
            appWFUri={appWFUri}
          />
        </Grid>
      )}
    </Grid>
  );
};

DeploymentDetails.propTypes = {
  deploymentDetails: PropTypes.object.isRequired,
  environment: PropTypes.object.isRequired,
  isExpandable: PropTypes.bool,
  onClickCallBack: PropTypes.func,
};

export default DeploymentDetails;
