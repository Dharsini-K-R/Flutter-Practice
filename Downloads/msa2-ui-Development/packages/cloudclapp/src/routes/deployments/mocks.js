import { DEPLOYMENT_VARIABLES_NAME } from "cloudclapp/src/Constants";

export const environments = [
  {
    envName: "Development",
    envId: 99,
    envReference: "68cee9b1-9226-4fdd-a4be-0e1e57977139",
    status: "OK",
    cclapOwner: 0,
    deployments: [
      {
        deploymentId: "126016",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "RUNNING",
      },
      {
        deploymentId: "126088",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "FAIL",
      },
      {
        deploymentId: "126018",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "ENDED",
      },
    ],
    envEntityId: 288,
  },
  {
    envName: "QA",
    envId: 118,
    envReference: "fd8016e0-d367-4179-956e-3f5f35a8d764",
    status: "NEVERREACHED",
    cclapOwner: 105,
    deployments: [
      {
        deploymentId: "126019",
        deploymentName: "Dev_My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "RUNNING",
      },
      {
        deploymentId: "126020",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "FAIL",
      },
      {
        deploymentId: "126021",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "ENDED",
      },
    ],
    envEntityId: 289,
  },
  {
    envName: "Staging",
    envId: 119,
    envReference: "fd8016e0-d367-4179-956e-3f5f35a8d764",
    status: "NEVERREACHED",
    cclapOwner: 105,
    deployments: [
      {
        deploymentId: "126022",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "RUNNING",
      },
      {
        deploymentId: "126023",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "FAIL",
      },
      {
        deploymentId: "126024",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "ENDED",
      },
    ],
    envEntityId: 289,
  },
];

export const applications = {
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_NAME]: "NATS0",
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_SLUG]: "NATS0",
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_VERSION]: "V2.1.9",
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_DESCRIPTION]:
    "Amazon Web Services provides Information Technology infrastructure services to businesses in the form of web services",
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_LOGO]:
    "https://d1q6f0aelx0por.cloudfront.net/product-logos/library-nats-logo.png",
};

export const environment = {
  envName: "hto env 24",
  envId: 126,
  envUbiqubeId: "005A126",
  description: "aslkd adsg aslgd ll",
  cloudVendor: "aws",
  cloudService: "eks",
  orgName: "hto Org extRef",
  envIp: "127.0.0.1",
  envHostname: "temp-host",
  username: "a",
  envReference: "88faa4e6-be22-444f-a085-e362273d76af",
  importFlag: false,
  serviceId: "123",
  serviceName: "Env",
  envWFUri:
    "Process/cloudclapp-wf/Provision_an_EKS_Cluster__AWS_/Provision_an_EKS_Cluster__AWS_",
  appWFUri:
    "Process/cloudclapp-wf/Provision_an_EKS_Apps__AWS_/Provision_an_EKS_Apps__AWS_",
  deploymentCount: 1,
  userCount: 1,
  scanData: null,
  cclapOwner: 105,
  envEntityId: 306,
  status: "Completed",
};
