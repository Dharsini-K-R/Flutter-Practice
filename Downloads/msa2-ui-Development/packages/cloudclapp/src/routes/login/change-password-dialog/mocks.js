export const linkValidationResponse = {
  message: "Link is valid",
};
