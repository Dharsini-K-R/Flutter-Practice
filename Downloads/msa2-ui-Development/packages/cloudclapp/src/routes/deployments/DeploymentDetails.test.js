import React from "react";
import { renderWithProvider } from "msa2-ui/src/utils/test-utils";
import DeploymentDetails from "./DeploymentDetails";
import initialState from "cloudclapp/src/store/initialState";
import { cloudVendors } from "cloudclapp/src/routes/dashboard/mock";
import { environment } from "cloudclapp/src/routes/environments/details/mocks";

const mockState = {
  ...initialState,
  designations: { ...initialState.designations, cloudVendors },
};

const deployment = {
  deploymentId: "126016",
  deploymentName: "My_Big_Shop_Test_4",
  status: "RUNNING",
  deploymentUser: "test-user",
};

const isExpandable = true;

describe("DeploymentDetails component", () => {
  it("renders correctly", () => {
    const { container, getByText } = renderWithProvider({
      childComponent: (
        <DeploymentDetails
          deploymentDetails={deployment}
          isExpandable={isExpandable}
          environment={environment}
        />
      ),
      mockState,
    });

    expect(container.firstChild).toMatchSnapshot();

    expect(getByText("Status:")).toBeTruthy();
  });
});
