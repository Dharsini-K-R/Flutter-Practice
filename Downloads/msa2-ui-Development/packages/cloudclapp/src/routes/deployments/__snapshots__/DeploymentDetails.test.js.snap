// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentDetails component renders correctly 1`] = `
<div
  class="MuiGrid-root makeStyles-boxWrapper MuiGrid-container"
>
  <div
    class="MuiGrid-root makeStyles-deploymentWrapper MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
  >
    <div
      class="makeStyles-titleContainer"
    >
      <span
        class="MuiBadge-root"
      >
        <i
          aria-hidden="true"
          class="la la-stream makeStyles-icon"
          role="presentation"
          style="font-size: 21px;"
        />
        <span
          class="MuiBadge-badge makeStyles-badge makeStyles-badge MuiBadge-anchorOriginBottomRightCircular MuiBadge-dot"
        />
      </span>
      <span
        class="makeStyles-label"
      />
      <p
        class="MuiTypography-root makeStyles-iconText MuiTypography-body1"
        id="DEPLOYMENTS_DEPLOYMENT_NAME_126016"
      >
        My_Big_Shop_Test_4
      </p>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-userContainer makeStyles-deploymentWrapper MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-2 MuiGrid-grid-lg-3"
  >
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-buttonRoot MuiButton-text makeStyles-icon MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      id="DEPLOYMENT_DETAILS_OWNER_BUTTON_126016"
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular makeStyles-avatar makeStyles-avatar MuiAvatar-colorDefault"
        >
          <i
            aria-hidden="true"
            class="la la-user-alt"
            role="presentation"
          />
        </div>
        <p
          class="MuiTypography-root makeStyles-name MuiTypography-body1"
        >
          Unknown User - test-user
        </p>
      </span>
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
    spacing-lg="2"
  >
    <div
      class="MuiGrid-root makeStyles-deploymentWrapper makeStyles-containerPlacement MuiGrid-container MuiGrid-item MuiGrid-grid-xs-true"
    >
      <p
        class="MuiTypography-root makeStyles-iconText MuiTypography-body1"
        id="DEPLOYMENTS_DEPLOYMENT_STATUS_126016"
      >
        Status
        :
        <span
          class="makeStyles-statusRunning makeStyles-statusRunning"
        >
          Running...
        </span>
      </p>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-1 MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-1"
      />
    </div>
    <div
      class="MuiGrid-root makeStyles-deploymentWrapper makeStyles-containerPlacement MuiGrid-container MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-1 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
    >
      <i
        aria-hidden="true"
        class="la la-angle-down makeStyles-expandCollapseIcon"
        id="DEPLOYMENTS_DEPLOYMENT_EXPAND_126016"
        role="presentation"
      />
    </div>
  </div>
</div>
`;
